### PR TITLE
[Display_01] Create Public Interfaces and Shared Data Types

### LINKED ISSUE #26: [Display_01] Create Public Interfaces and Shared Data Types
**What is the use case? Please describe.**
Create the public Java interface for each interface in the `edu.oakland.production.display01` package, and the public methods for every data type in the `edu.oakland.helper.display01` package. Note that this is part of sprint deliverable task 2.3.2.

**Setup**

To be done by @EC01EC , @stevecatherman , or @casher-ou before the rest of the team starts working on it:

1. Use the documentation at https://github.com/andrewdimmer/csi-3370-software-project/tree/display_01-production-uml-class-diagrams/docs/uml-class-diagrams/display01 to identify the classes and interfaces to create.
2. Edit this issue message to add a checkbox for each class or interface, and assign it to the correct team member.

**Steps**

To be done by each team member for each class or interface they are creating:

1. Checkout and/or pull the branch `display01-public-interfaces`
2. Make the code changes to create the class or interface described in https://github.com/andrewdimmer/csi-3370-software-project/tree/display_01-production-uml-class-diagrams/docs/uml-class-diagrams/display01
3. Test your changes locally with `gradle build` if you have Gradle installed, or test it with `./gradlew build` or `.\gradlew.bat build` if you do not.
4. Commit your changes with the message `[Display_01] Created <<<Class/Interface_Name_Here>>>`
5. Push your changes
6. If you are the first person to push, create a pull request draft with the name `[Display_01] Create Public Interfaces and Shared Data Types` and the rest of the information like description, tags, etc. For reference, the settings on the right hand side should match this issue, with the exception that the assignees on the PR are just those who are pushing code.
7. If you are the last person to push, release the pull request for reviews, and add @andrewdimmer as a reviewer.

**Tasks to Complete**

- [x] Assign Tasks - @EC01EC , @stevecatherman , or @casher-ou
- [x] Create DisplayGPSInterface - @reedsikorskiOU or @DLidz 
- [x] Create DisplayGPSManager - @atpoynter or @elilley15
- [x] Create DisplayGPSReciever - @EC01EC or @casher-ou
- [x] Create SatelliteSignalCheckRequest - @stevecatherman or @John-A11 

**Additional context**
Note: if for some reason you need to change a public interface or shared data type, reach out to @andrewdimmer who can help you manage making that BREAKING CHANGE.

### PR DESCRIPTION
**Pull request by reedsikorskiOU**
Closes #26 

Created DisplayGpsInterface.java and DisplayGpsInterfaceImplementation.java preGradle Testing

### CODE REVIEW

File: src/main/java/edu/oakland/production/display01/DisplayGpsInterface.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package edu.oakland.production;
```
Comment (andrewdimmer) [inline src/main/java/edu/oakland/production/display01/DisplayGpsInterface.java]:
```suggestion
package edu.oakland.production.display01;
```

File: src/main/java/edu/oakland/production/display01/DisplayGpsInterfaceImplementation.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
+package edu.oakland.production;
```
Comment (andrewdimmer) [inline src/main/java/edu/oakland/production/display01/DisplayGpsInterfaceImplementation.java]:
```suggestion
package edu.oakland.production.display01;
```

File: src/main/java/edu/oakland/production/display01/DisplayGpsManager.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package edu.oakland.production;
```
Comment (andrewdimmer) [inline src/main/java/edu/oakland/production/display01/DisplayGpsManager.java]:
```suggestion
package edu.oakland.production.display01;
```

File: src/main/java/edu/oakland/production/display01/DisplayGpsManagerImplementation.java [hunk 4]
```diff
@@ -0,0 +1,17 @@
+package edu.oakland.production;
```
Comment (andrewdimmer) [inline src/main/java/edu/oakland/production/display01/DisplayGpsManagerImplementation.java]:
```suggestion
package edu.oakland.production.display01;
```

File: src/main/java/edu/oakland/production/display01/DisplayGpsManagerImplementation.java [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+import edu.oakland.helper.admin.Satellite;
+import edu.oakland.helper.display01.SatelliteSignalCheckRequest;
+
+public class DisplayGpsManagerImplementation implements DisplayGpsManager{
```
Comment (andrewdimmer) [inline src/main/java/edu/oakland/production/display01/DisplayGpsManagerImplementation.java]:
```suggestion
public class DisplayGpsManagerImplementation implements DisplayGpsManager {
```

File: src/main/java/edu/oakland/production/display01/DisplayGpsInterface.java [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+
+import edu.oakland.helper.admin.Satellite;
+
+public interface DisplayGpsInterface {
+
+  public void recieveGpsSingnals(Satellite satelliteSignal);
```
Comment (andrewdimmer) [inline src/main/java/edu/oakland/production/display01/DisplayGpsInterface.java]:
```suggestion
  public void receiveGpsSignals(Satellite satelliteSignal);
```

File: src/main/java/edu/oakland/production/display01/DisplayGpsInterfaceImplementation.java [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+
+import edu.oakland.helper.admin.Satellite;
+
+public class DisplayGpsInterfaceImplementation implements DisplayGpsInterface {
+
+  public void recieveGpsSingnals(Satellite satelliteSignal) {
```
Comment (andrewdimmer) [inline src/main/java/edu/oakland/production/display01/DisplayGpsInterfaceImplementation.java]:
```suggestion
  public void receiveGpsSignals(Satellite satelliteSignal); {
```

File: src/main/java/edu/oakland/production/display01/DisplayGpsManager.java [hunk 8]
```diff
@@ -0,0 +1,12 @@
...
+
+import edu.oakland.helper.admin.Satellite;
+import edu.oakland.helper.display01.SatelliteSignalCheckRequest;
+
+public interface DisplayGpsManager {
+
+  public String receiveGpsSignalStrengths(Satellite satelliteSignal);
```
Comment (andrewdimmer) [inline src/main/java/edu/oakland/production/display01/DisplayGpsManager.java]:
```suggestion
  public String receiveGpsSignalStrength(Satellite satelliteSignal);
```

File: src/main/java/edu/oakland/production/display01/DisplayGpsManagerImplementation.java [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+    
+  public String receiveGpsSignalStrengths(Satellite satelliteSignal) {
```
Comment (andrewdimmer) [inline src/main/java/edu/oakland/production/display01/DisplayGpsManagerImplementation.java]:
```suggestion
  public String receiveGpsSignalStrength(Satellite satelliteSignal) {
```

**Review by andrewdimmer — CHANGES_REQUESTED** [review]
Couple of spelling changes, couple of package issues, and a small  breaking change to the UML model based on a typo before we are ready to merge this one.

File: src/main/java/edu/oakland/production/display01/DisplayGpsInterface.java [hunk 10]
```diff
@@ -0,0 +1,15 @@
+package edu.oakland.production.display01;
+
+import edu.oakland.helper.admin.Satellite;
+
+public interface DisplayGpsInterface {
+
+  public void recieveGpsSignals(Satellite satelliteSignal);
```
Comment (andrewdimmer) [inline src/main/java/edu/oakland/production/display01/DisplayGpsInterface.java]:
```suggestion
  public void receiveGpsSignal(Satellite satelliteSignal);
```

File: docs/uml-class-diagrams/display01/production/DisplayGpsInterface/DisplayGpsInterface.puml [hunk 11]
```diff
@@ -4,7 +4,7 @@ package edu.oakland.production.display01 {
     interface DisplayGpsInterface {
         __
         .. Use Case 1 ..
-        {abstract} + void revieveGPSSignals(Satellite satelliteSignal)
+        {abstract} + void recieveGpsSignal(Satellite satelliteSignal)
```
Comment (andrewdimmer) [inline docs/uml-class-diagrams/display01/production/DisplayGpsInterface/DisplayGpsInterface.puml]:
```suggestion
        {abstract} + void receiveGpsSignal(Satellite satelliteSignal)
```

File: src/main/java/edu/oakland/production/display01/DisplayGpsInterfaceImplementation.java [hunk 12]
```diff
@@ -0,0 +1,23 @@
+package edu.oakland.production.display01;
+
+import edu.oakland.helper.admin.Satellite;
+
+public class DisplayGpsInterfaceImplementation implements DisplayGpsInterface {
+
+  public void recieveGpsSignals(Satellite satelliteSignal) {
```
Comment (andrewdimmer) [inline src/main/java/edu/oakland/production/display01/DisplayGpsInterfaceImplementation.java]:
```suggestion
  public void receiveGpsSignal(Satellite satelliteSignal) {
```

**Review by andrewdimmer — CHANGES_REQUESTED** [review]
Few more typos, should be good to merge once these are fixed though!

**Review by andrewdimmer — APPROVED** [review]
Looks good to me, you're all set to merge!

**Review by EC01EC — APPROVED** [review]
looks good

**Pull request merged**